elements/lisk-p2p/src/peer/inbound.ts

```diff
@@ -46,12 +56,25 @@ export class InboundPeer extends Peer {
 			this.emit(EVENT_INBOUND_SOCKET_ERROR, error);
 		};
 		this._handleInboundSocketClose = (code, reason) => {
+			if (this._pingTimeoutId) {
+				clearTimeout(this._pingTimeoutId);
+			}
 			this.emit(EVENT_CLOSE_INBOUND, {
 				peerInfo,
 				code,
 				reason,
 			});
 		};
+		this._pingStart = Date.now();
```

Comment:
This `_pingStart ` is not needed here. It should be defined as a scoped `const pingStart` inside the `_sendPing` function (no `this` needed); keeping it only inside the scope of the `_sendPing` function would guarantee that it cannot be mixed up with a different ping. It's possible that our node could be waiting for the response from multiple parallel pings at the same time so the calculation for each ping/pong must have its own independent scope.